api/scpca_portal/test/management/commands/test_load_data.py

```diff
@@ -54,8 +59,29 @@ def assertProjectData(self, project):
         self.assertIsNotNone(sample.tissue_location)
         self.assertIsNotNone(sample.treatment)
 
-    def assertProjectReadmeContains(self, text, project_zip):
-        self.assertIn(text, project_zip.read("README.md").decode("utf-8"))
+    def assertProjectReadmeContent(self, zip_file):
+        def get_masked_content(content):
+            """
+            Replace the project ID and date with placeholders for format testing.
+            """
+            content = re.sub(r"SCPCP\d{6}", "SCPCPXXXXXX", content)
+            content = re.sub(
+                r"Generated on: \d{4}-\d{2}-\d{2}", "Generated on: YYYY-MM-DD", content
+            )
+
+            return content
+
+        if README_FILE in zip_file.namelist():
+            # Get the corresponding saved readme output path based on the zip filename
+            readme_filename = re.sub(r"^[A-Z\d]+_", "", Path(zip_file.filename).stem) + ".md"
+            saved_readme_output_path = README_DIR / readme_filename
+
+            with zip_file.open(README_FILE) as readme_file:
+                output_content = get_masked_content(readme_file.read().decode("utf-8").strip())
+            with saved_readme_output_path.open("r", encoding="utf-8") as saved_readme_file:
+                expected_content = get_masked_content(saved_readme_file.read().strip())
+
+            self.assertEqual(expected_content, output_content)
```

Comment:
Let's add a custom message here so we know which readme template we are asserting against.

Reply:
Try converting both contents to lists before asserting equality to help determine the line in which the files are different. I think `expected_content.splitlines()` (same for output_content) is what we want so that we can see difference per line without doing maxDiff = None

Reply:
See at 09eed514808eb4537f50e1d6da23c34045835049.